### ISSUE TITLE
[Bug]: Accuracy bug of ulysses on Qwen-image-edit

### ISSUE THREAD

**Issue by jiangmengyu18** [vllm-project/vllm-omni#347]
### Your current environment

Hardware: Ascend G8600
Model: Qwen-Image-Edit-2509

### 🐛 Describe the bug

Referencing https://github.com/vllm-project/vllm-omni/pull/330, the main branch was adapted for Qwen-Image-Edit-2509. During the testing of Ulysses, it was discovered that the result accuracy was incorrect, with the most severe issues occurring on the 8-card configuration.

Preliminary analysis：
Documents:`vllm_omni/diffusion/models/qwen_image/qwen_image_transformer.py:625-628`
```python
if self.parallel_config.sequence_parallel_size > 1:
    hidden_states = torch.chunk(hidden_states, get_sequence_parallel_world_size(), dim=-2)[
        get_sequence_parallel_rank()
    ]
```
Only the image hidden_states were chunked, while the text was not chunked.

Documents:`vllm_omni/diffusion/models/qwen_image/qwen_image_transformer.py:323-332`
```python
# Concatenate for joint attention
# Order: [text, image]
joint_query = torch.cat([txt_query, img_query], dim=1)
joint_key = torch.cat([txt_key, img_key], dim=1)
joint_value = torch.cat([txt_value, img_value], dim=1)

# Compute joint attention
joint_hidden_states = self.attn(
    joint_query,
    joint_key,
    joint_value,
)
```
txt_qkv and img_qkv are concatenated and fed into self.attn.

Documents:`vllm_omni/diffusion/attention/layer.py:96-98`
```python
# scatter 2, gather 1
# (bs, seq_len/N, head_cnt, head_size) -> (bs, seq_len, head_cnt/N, head_size)
q = SeqAllToAll4D.apply(self.ulysses_pg, query, self.scatter_idx, self.gather_idx, self.use_sync)
k = SeqAllToAll4D.apply(self.ulysses_pg, key, self.scatter_idx, self.gather_idx, self.use_sync)
v = SeqAllToAll4D.apply(self.ulysses_pg, value, self.scatter_idx, self.gather_idx, self.use_sync)
```
Since qkv is formed by concatenating txt and img, the result of all_to_all here is similar to: [txt, img_sp0, txt, img_sp1, txt, img_sp2, txt, img_sp3]. This is incorrect. The correct result should be similar to: [txt, img_sp0, img_sp1, img_sp2, img_sp3], due to only image hidden_states deploying sp.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.

**Comment by hsliuustc0106** [comment]
@wtomin please check whether this bug also happens in GPU

**Comment by ZJY0516** [comment]
@jiangmengyu18 Could you please provide a repor script?

**Comment by ZJY0516** [comment]
The generated image appears normal.

```
python image_edit.py --image ./coffee.png --prompt "Put some cube sugar nearby the coffee cup" --output ./coffee-edit-0.png --ulysses_degree 8
```

<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/35f5c8a8-5e47-4735-8826-9091075b481f" />

**Comment by jiangmengyu18** [comment]
@ZJY0516 
The case I tested is:
prompt: Make Pikachu hold a sign that says 'Qwen Edit is awesome', yarn art style, detailed, vibrant color
image: https://modelers.cn/models/MindIE/Qwen-Image-Edit-2509/blob/main/yarn-art-pikachu.png

```python
python  examples/offline_inference/image_to_image/image_edit.py \
  --image ./yarn-art-pikachu.png \
  --prompt "Make Pikachu hold a sign that says 'Qwen Edit is awesome', yarn art style, detailed, vibrant color" \
  --seed 42 \
  --output output_image_edit.png \
  --num_inference_steps 50 \
  --cfg_scale 4.0 \
  --cache_backend cache_dit \
  --ulysses_degree 8
```

**Comment by jiangmengyu18** [comment]
You can test some more complex cases, for example, one that includes some text.

**Comment by wtomin** [comment]
Thanks for pointing it out. It is indeed problematic. As you mentioned, for joint qkv, we need sepecial handlings to seperate the text embedding from image embedding before feeding it to self.attn. I will make another PR to solve it ASAP.

**Comment by ZJY0516** [comment]
I think this is what we need to catch in CI

**Comment by jiangmengyu18** [comment]
@wtomin 
I made a temporary revision to the code; you can take a look for reference.
https://github.com/Fishermanykx/vllm-omni/commit/85e932a241a03c5b8c43c8e4c1418f2165bd2cc3

**Comment by wtomin** [comment]
> [@wtomin](https://github.com/wtomin) I made a temporary revision to the code; you can take a look for reference. [Fishermanykx@85e932a](https://github.com/Fishermanykx/vllm-omni/commit/85e932a241a03c5b8c43c8e4c1418f2165bd2cc3)

Thanks. I can synchronize your edits to #377 , but I don't have the Ascend NPU env to test.

**Comment by wtomin** [comment]
This issue can be closed after #377
